packages/flutter_tools/lib/src/isolated/resident_web_runner.dart

```diff
@@ -254,15 +255,38 @@ class ResidentWebRunner extends ResidentRunner {
 
     try {
       return await asyncGuard(() async {
+        Future<int> getPort() async {
```

Comment:
Does this need to be a local function, since it's only invoked once? Can we just inline this code?

Reply:
It doesn't need to be a (local) function, but I don't think a function needs to be called more than once to justify its existence. In this case, I went with a function to

* provide a clear start and end to port selection logic,
* limit the scope of port selection logic, making it more clear that it doesn't interact with any other code within the `asyncGuard` callback, and
* make it easier for first-time readers of the `run` method by moving the low-level details of port selection out of the way (you can fold/hide `getPort` and still understand what `run` does)

amongst other things.